Makefile

```diff
@@ -1,18 +1,124 @@
-all: build
-.PHONY: all
+# Old-skool build tools.
+#
+# Targets (see each target for more information):
+#   all: Build code.
+#   build: Build code.
+#   check: Run verify, build, unit tests and cmd tests.
+#   test: Run all tests.
+#   run: Run all-in-one server
+#   clean: Clean up.
 
-build:
-	go build -o _output/bin/dockerregistry github.com/openshift/image-registry/cmd/dockerregistry
-.PHONY: build
+OUT_DIR = _output
+OS_OUTPUT_GOPATH ?= 1
 
-build-image: build
-	hack/build-image.sh
-.PHONY: build-image
+export GOFLAGS
+export TESTFLAGS
+# If set to 1, create an isolated GOPATH inside _output using symlinks to avoid
+# other packages being accidentally included. Defaults to on.
+export OS_OUTPUT_GOPATH
+# May be used to set additional arguments passed to the image build commands for
+# mounting secrets specific to a build environment.
+export OS_BUILD_IMAGE_ARGS
 
-verify: build
-	go test github.com/openshift/image-registry/pkg/...
+# Tests run using `make` are most often run by the CI system, so we are OK to
+# assume the user wants jUnit output and will turn it off if they don't.
+JUNIT_REPORT ?= true
+
+# Build code.
+#
+# Args:
+#   WHAT: Directory names to build.  If any of these directories has a 'main'
+#     package, the build will produce executable files under $(OUT_DIR)/local/bin.
+#     If not specified, "everything" will be built.
+#   GOFLAGS: Extra flags to pass to 'go' when building.
+#   TESTFLAGS: Extra flags that should only be passed to hack/test-go.sh
+#
+# Example:
+#   make
+#   make all
+#   make all WHAT=cmd/oc GOFLAGS=-v
+all build:
```

Comment:
why doesn't `make all` build rpms and images?  (and run verify for that matter).

having "all" be an alias for "build" seems weird.

Reply:
Convention today, we can talk about changing it but not here.